org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/GradleDistribution.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2018 the original author or authors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.eclipse.buildship.core;
+
+import java.io.File;
+import java.net.URI;
+
+/**
+ * Represents a Gradle distribution.
+ *
+ * @author Donat Csikos
+ * @since 3.0
+ * @noimplement this interface is not intended to be implemented by clients
+ */
+public abstract class GradleDistribution {
```

Comment:
We should have another chat about this class on Monday. Right now it is basically useless for our end users, so we might es well hide it completely (like ProjectConnection does).

On the other hand, if we want to allow users to build their own configuration UI for instance, we should fully expose the different subtypes.